module/functions/azure/aad/Get-AzureAdDirectoryObject.Tests.ps1

```diff
@@ -0,0 +1,202 @@
+                Assert-MockCalled _userById -Times 0
+                Assert-MockCalled _groupByName -Times 1
+                Assert-MockCalled _spByName -Times 1
+                Assert-MockCalled _userByName -Times 1
+            }
+        }
+    }
+
+    Context "Finding a service principal" {
+
+        Context "Searching by ObjectId" {
+            $global:methodUsed = $null
+            $res = Get-AzureAdDirectoryObject -Criterion $mockServicePrincipal.ObjectId
+            
+            It "should return the group" {
+                $methodUsed | Should -Be "ObjectId"
+                $res.DisplayName | Should -Be $mockServicePrincipal.DisplayName
+                
+                Assert-MockCalled _groupById -Times 1
+                Assert-MockCalled _spByAppId -Times 1
+                Assert-MockCalled _spByObjectId -Times 1
+                Assert-MockCalled _userById -Times 1
+                Assert-MockCalled _groupByName -Times 0
+                Assert-MockCalled _spByName -Times 0
+                Assert-MockCalled _userByName -Times 0
+            }
+        }
+
+        Context "Searching by ApplicationId" {
+            $global:methodUsed = $null
+            $res = Get-AzureAdDirectoryObject -Criterion $mockServicePrincipal.ApplicationId
+            
+            It "should return the group" {
+                $methodUsed | Should -Be "ApplicationId"
+                $res.DisplayName | Should -Be $mockServicePrincipal.DisplayName
+                
+                Assert-MockCalled _groupById -Times 1
+                Assert-MockCalled _spByAppId -Times 1
+                Assert-MockCalled _spByObjectId -Times 1
+                Assert-MockCalled _userById -Times 1
+                Assert-MockCalled _groupByName -Times 0
+                Assert-MockCalled _spByName -Times 0
+                Assert-MockCalled _userByName -Times 0
+            }
+        }
+
+        Context "Searching by DisplayName" {
+            $global:methodUsed = $null
+            $res = Get-AzureAdDirectoryObject -Criterion $mockServicePrincipal.DisplayName
+            
+            It "should return the group" {
+                $methodUsed | Should -Be "DisplayName"
+                $res.ObjectId | Should -Be $mockServicePrincipal.ObjectId
+                
+                Assert-MockCalled _groupById -Times 0
+                Assert-MockCalled _spByAppId -Times 0
+                Assert-MockCalled _spByObjectId -Times 0
+                Assert-MockCalled _userById -Times 0
+                Assert-MockCalled _groupByName -Times 1
+                Assert-MockCalled _spByName -Times 1
+                Assert-MockCalled _userByName -Times 1
+            }
+        }
+    }
+
+    Context "Finding a user" {
+
+        Context "Searching by ObjectId" {
+            $global:methodUsed = $null
+            $res = Get-AzureAdDirectoryObject -Criterion $mockUser.ObjectId
+            
+            It "should return the group" {
```

Comment:
Similar to Ian's comment above, I think these assertions should be `It "should return the user"`

Reply:
Updated